opm/simulators/wells/WellGroupHelpers.cpp

```diff
@@ -903,7 +906,20 @@ namespace WellGroupHelpers
     #endif
                     } else {
                         // Table number specified as 9999 in the deck, no pressure loss.
-                        node_pressures[node] = up_press;
+                        if (network.node(node).as_choke()){
+                            // Node pressure is set to the common THP of the wells
+                            const auto up_branch = network.uptree_branch(node);
+                            assert(up_branch);
+                            const auto up_pressure = node_pressures[(*up_branch).uptree_node()];
+                            const auto group_thp = group_state.well_group_thp(node);
+                            if (group_thp >= up_pressure){
```

Comment:
This can be simplified as 
```c++
node_pressures[node] = group_thp >= up_pressure ? group_thp : up_press;
```

A comment here to tell for a production network, the choke pressure can not be smaller than zero, so group_thp can not be lower than up_press. Something similar to that.

Reply:
That is indeed much neater. I changed it according to your suggestion and added a comment.